solarforecastarbiter/metrics/calculator.py

```diff
@@ -337,38 +334,25 @@ def calculate_probabilistic_metrics(processed_fx_obs, categories, metrics):
     elif len(metrics) == 0:
         raise RuntimeError("No metrics specified.")
 
-    # Separate metrics by type
-    single_metrics = list(set(metrics) - set(probabilistic._REQ_DIST))
-    dist_metrics = list(set(metrics) & set(probabilistic._REQ_DIST))
-
-    # Single (per ProbabilisticForecastConstantValue) forecast metrics
-    if single_metrics:
-        cvs = processed_fx_obs.original.forecast.constant_values
-        for orig, ref, cv in zip(fx_fx_prob, ref_fx_fx_prob, cvs):
-            single_out = copy.deepcopy(shared_dict)
-            single_out['forecast_id'] = cv.forecast_id
-            results = _calculate_probabilistic_metrics_from_df(
-                _create_prob_dataframe(obs, [orig], [ref]),
-                categories, single_metrics, cv.interval_label)
-            single_out['values'] = _sort_metrics_vals(
-                results, datamodel.ALLOWED_PROBABILISTIC_METRICS)
-            single_cv_results.append(datamodel.MetricResult.from_dict(
-                single_out))
-
-    # Distribution (per ProbabilisticForecast) forecast metrics
-    if dist_metrics:
-        dist_out = copy.deepcopy(shared_dict)
-        dist_out['forecast_id'] = processed_fx_obs.original.forecast.forecast_id  # NOQA
-        results = _calculate_probabilistic_metrics_from_df(
-            _create_prob_dataframe(obs, fx_fx_prob, ref_fx_fx_prob),
-            categories,
-            dist_metrics,
-            processed_fx_obs.original.forecast.interval_label)
-        dist_out['values'] = _sort_metrics_vals(
-            results, datamodel.ALLOWED_PROBABILISTIC_METRICS)
-        dist_result = datamodel.MetricResult.from_dict(dist_out)
-
-    return (single_cv_results, dist_result)
+    fx = processed_fx_obs.original.forecast
+    out['forecast_id'] = fx.forecast_id
+
+    # Determine proper metrics by instance type
```

Comment:
note this behavior in the docstring, either in the intro section or in the returns section.

either way the returns section of the docstring needs updating.

Reply:
Updated docstrings